Utilities/AzureOnboardingCLI/azure-automatic-role-assignment.py

```diff
@@ -175,16 +201,25 @@ def set_azure_credentials(
     Returns:
         bool: True if credentials were set successfully, False otherwise
     """
-    session = SpotinstSession(auth_token=token, base_url=SPOT_API_BASE_URL)
-    client = session.client("setup_azure")
-    client.account_id = account_id
-    azure_credentials = AzureCredentials(
-        client_id=client_id,
-        client_secret=client_secret,
-        tenant_id=tenant_id,
-        subscription_id=subscription_id,
-    )
-    return client.set_credentials(azure_credentials)
+    log(f"Updating linked credentials for Spot account {account_id}")
+    try:
+        session = SpotinstSession(auth_token=token, base_url=SPOT_API_BASE_URL)
+        client = session.client("setup_azure", timeout=15)
+        client.account_id = account_id
+        azure_credentials = AzureCredentials(
+            client_id=client_id,
+            client_secret=client_secret,
+            tenant_id=tenant_id,
+            subscription_id=subscription_id,
+        )
+        result = client.set_credentials(azure_credentials)
+    except SpotinstClientException as e:
```

Comment:
Fine but seems like this would retry if the credentials were failing for some other "valid" reason. That should hopefully not happen. But does the 2nd attempt's error (if any) get logged?

Reply:
Yes it will get logged by the caller's try-catch.